src/binary-reader-ir.cc

```diff
@@ -1273,6 +1301,105 @@ Result BinaryReaderIR::OnFunctionName(Index index, string_view name) {
   return Result::Ok;
 }
 
+Result BinaryReaderIR::SetTableName(Index index, string_view name) {
+  if (name.empty()) {
+    return Result::Ok;
+  }
+  if (index >= module_->tables.size()) {
+    PrintError("invalid table index: %" PRIindex, index);
```

Comment:
Turns out that templatizing the error message here is not easy in C++... https://stackoverflow.com/questions/1826464/c-style-strings-as-template-arguments

I'll keep trying but will probably land as is and followup with a templetization.

Reply:
Could we template on a `typename T`, then pass the entire string `"Invalid table index: %" PRIindex` as an arg, and maybe let some deduction/conversion happen?
Follow-up (with a TODO to indicate what we would like to do here) would be totally fine.